core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -247,14 +247,13 @@ private void parseArguments(final String[] args) {
             .describedAs("file name");
         forceOption = optionParser.accepts("force", "Force the removal of members of the consumer group (intended to remove stopped members if a long session timeout was used). " +
                 "Make sure to shut down all stream applications when this option is specified to avoid unexpected rebalances.");
-        executeOption = optionParser.accepts("execute", "Execute the command.");
+
+        // TODO: deprecated in 3.0 can be removed in a future release -- https://issues.apache.org/jira/browse/KAFKA-7606
+        executeOption = optionParser.accepts("execute", "This option is deprecated and will be removed in a future release; it is not required to specify this in order to execute the command");
```

Comment:
We did not to do this for ZK flag, but should we also log/print if the flag is used?

Reply:
done

Reply:
I tend to agree. -- My main point was that I wanted to raise awareness and that we should make a conscious decision about it.

Might also be good to hear from @guozhangwang about it.

Reply:
Some weird things I noticed: First, the ticket says `--execute` was deprecated in 1.1, but it doesn't even exist in 1.0 (or earlier AFAICT but I did not check all branches). Did we...add it and deprecate it in the same release? Huh?

Also, I just read KIP-171 and it's honestly super confusing. Regarding the deprecation, it says `"Current option '–execute' will be marked as "going-to-be-deprecated" and replaced by 'dry-run' parameter, until it gets replaced."`

Not sure how to interpret "going-to-be-deprecated" -- does that mean we actually did not deprecate this flag in 1.1 but just intend to do so at a later time?

Reply:
It's a long time ago... Cannot recall. Because it's many year back, might also not be worth to spent time digging into to to reconstruct it?